packages/eddsa-pcd/src/EdDSAPCD.ts

```diff
@@ -6,62 +10,175 @@ import {
+ * 
+ * @summary The arguments for an EdDSA PCD.
+ * @interface
+ */
 export interface EdDSAPCDArgs {
-  // The 32-byte EdDSA private key to sign the message with, as a hex string.
-  // Any 32 bytes work as a private key. Use {@link newEdDSAPrivateKey} to
-  // generate a new one securely.
+  /**
+   * Any 32-bytes EdDSA private key (32-bytes hexadecimal string) for signing the message.
+   * Use {@link newEdDSAPrivateKey} to generate a new one securely.
+   * 
+   * @type {StringArgument}
+   */
   privateKey: StringArgument;
-  // The message is an array of bigints, represented here as an array of
-  // strings
+  
+  /**
+   * An array of contents to be signed with the private key. 
+   * The message is represented as a string version of a set of BigInts signing object.
+   * 
+   * @type {StringArrayArgument}
+   */
   message: StringArrayArgument;
-  // A unique string identifying the PCD
+ 
+  /**
+   * The unique identifier of the PCD.
+   * 
+   * @type {StringArgument}
+   */
   id: StringArgument;
 }
 
-// The claim is that we have a message that was signed by a public key
+/**
+ * This interface defines the arguments to make an EdDSA PCD claim.
+ * The claim contains a message signed with the private key corresponding to the given public key.
+ * 
+ * @summary The arguments for an EdDSA PCD claim.
+ * @interface
+ */
 export interface EdDSAPCDClaim {
-  // The public key is a pair of hex strings, representing a point on
-  // the elliptic curve
+  /**
+   * An EdDSA public key corresponding to the EdDSA private key used 
+   * for signing the message.
+   *
+   * @type {EDdSAPublicKey}
+   */
   publicKey: EDdSAPublicKey;
-  // The message is an array of bigints, each representing a field
-  message: bigint[];
+  
+  /**
+   * An array of contents signed with the private key.
+   * 
+   * @type {Array<bigint>}
+   */
+  message: Array<bigint>;
 }
 
-// The proof is the signature, in the form of a string representation of
-// hex digits.
-// This signature proves that the private key matching publicKey in
-// the claim was used to sign the message.
+/**
+ * This interface defines the arguments to make an EdDSA PCD proof.
+ * The proof is the signature which proves that the private key corresponding to the
+ * public key in the claim, has been successfully used to sign the message.
+ * 
+ * @summary The arguments for an EdDSA PCD proof.
+ * @interface
+ */
 export interface EdDSAPCDProof {
+  /**
+   * The EdDSA signature of the message.
+   * 
+   * @type {string}
+   */
   signature: string;
 }
 
+/**
+ * Create a new EdDSA PCD from an EdDSA PCD claim and proof.
+ * 
+ * @class 
+ * @classdesc This class represents an EdDSA PCD.
+ * @implements {PCD<EdDSAPCDClaim, EdDSAPCDProof>}
+ */
 export class EdDSAPCD implements PCD<EdDSAPCDClaim, EdDSAPCDProof> {
+  /** @public */
   type = EdDSAPCDTypeName;
+
+  /** @public */
+  id: string;
+  /** @public */
   claim: EdDSAPCDClaim;
+  /** @public */
   proof: EdDSAPCDProof;
```

Comment:
rather than commenting these are public, let's explicitly specify that they are public via typescipt.